### PR TITLE
Separate namespace for modules providing web interface

### PR DESCRIPTION
**Pull request by ikedas**
Following modules are referred only by web applications (wwsympa.fcgi and sympa_soap_server.fcgi), and they would be moved to separate namespace. Because:
  - Packager may select necessary modules.
  - On the next major release, web interface will be replaced entirely and developers may omit following modules.

| Current name | New name |
| --- | --- |
| Sympa::Auth | WWSympa::Auth |
| Sympa::Marc | WWSympa::Marc |
| Sympa::Marc::Search | WWSympa::Marc::Search |
| Sympa::Report | WWSympa::Report |
| Sympa::SOAP | WWSympa::SOAP |
| Sympa::SOAP::Transport | WWSympa::SOAP::Transport |
| Sympa::Session | WWSympa::Session |
| Sympa::SharedDocument | WWSympa::SharedDocument |
| Sympa::Tools::WWW | WWSympa::Tools |

### CODE REVIEW

File: src/sbin/task_manager.pl.in [hunk 1]
```diff
@@ -436,9 +436,9 @@ sub create {
     } else {
         $object = '_global';
         $task_file =
-              $spool_task . '/'
-            . $date . '.'
-            . $label . '.'
+              $spool_task . '/' 
+            . $date . '.' 
+            . $label . '.'
```
Comment (xavierba) [inline src/sbin/task_manager.pl.in]: This hunk introduces trailing white spaces.

Comment (ikedas) [reply src/sbin/task_manager.pl.in]: Maybe a bug of perltidy. I fixed. Thanks.

**Comment by racke** [comment]
Hm, I think the namespace "Sympa::WWW" would be better. Just my two cents.

**Comment by ikedas** [comment]
I think naming method is matter of preference.  The method getting more support would be carried out.

**Comment by racke** [comment]
So let's discuss the new namespace on the (developers) mailing list. I also think it is a good idea and it is a good time to this before sympa-7.0 branch starts to take off.

**Comment by ikedas** [comment]
There seems no objection to moving. Rest of issue is how we may move (rename). Before we move to dev list, we would be better to present rationale by each.

My favorite is new top-level namespace `WWSympa`. By this proposal, directory listing of `src/lib` will be:
``` bash
Sympa.pm
Sympa/
WWSympa.pm
WWSympa/
```
(`Conf.pm will be removed --- hopefully in this year, besides)

Separate namespaces will make separating packages easy:
  - Packagers may pick out `Sympa*` for core package and `WWSympa*` for web package.
  - Developers of new web interace may create new namespace and may work only on `Sympa` and `MojoSympa` namespaces.
  - I personally have plan of LMTP frontend of Sympa and I may work on `LMTPSympa` namespace.

etc.

**Comment by racke** [comment]
My favourite is to keep all our modules within a single namespace Sympa - this makes more sense to me and doesn't look so ugly like "WWSympa", "LMTPSympa", e.g.

- Sympa::WWW or Sympa::Web
- Sympa::LMTP or Sympa::Mail::LMTP

This doesn't affect the separation of the modules for packaging. 

Also, while we are at it, a separate namespace for 3rd party modules would be nice to, e.g. one of these:

- Sympa::Plugin::
- Sympa::Module::
- Sympa::Extension::

**Comment by ikedas** [comment]
Okey, let's continue on [dev list](https://listes.renater.fr/sympa/arc/sympa-developpers).

**Comment by eiro** [comment]
@dverdin both @racke  and i are agree about the namespaces @racke proposed. no one else provided another solution. what's your opinion ?

**Pull request closed without merging**